services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/crossregion/utils/CrossRegionUtils.java

```diff
@@ -55,11 +62,17 @@ public static boolean isS3RedirectException(Throwable exception) {
     }
 
     private static boolean isRedirectError(S3Exception exceptionToBeChecked) {
-        int statusCode = exceptionToBeChecked.statusCode();
-        return statusCode == REDIRECT_STATUS_CODE || statusCode == TEMPORARY_REDIRECT_STATUS_CODE;
+        if (REDIRECT_STATUS_CODES.stream().anyMatch(status -> status.equals(exceptionToBeChecked.statusCode()))) {
```

Comment:
For readibilty and debugging using separate ifs instead of clubbing to single statement with ors (||s)